packages/lib/src/vis/heatmap/ColorBar.tsx

```diff
@@ -83,10 +85,16 @@ function ColorBar(props: Props) {
             numTicks={adaptedNumTicks(gradientLength)}
             tickClassName={styles.tick}
             tickComponent={Tick}
-            tickFormat={axisScale.tickFormat(
-              adaptedNumTicks(gradientLength),
-              '.3~g'
-            )}
+            tickFormat={
+              scaleType === ScaleType.Log
+                ? axisScale.tickFormat(
```

Comment:
Calling `axisScale.tickFormat` with `.3~g` (i.e. with a precision) is pointless for "linearish" scales (i.e. other than log), since the function's purpose is to compute the precision dynamically.

For log scales, however, it returns empty labels for some ticks to avoid overlap: https://github.com/d3/d3-scale/blob/main/src/log.js#L118-L122. I might try to bring this logic into the app at some point to make this behaviour more explicit, to avoid the `@ts-ignore` below, and to maybe apply the same behaviour for symlog scales, which tend to have overlapping ticks as well.

Reply:
I don't understand the TS issue here. Is it a problem in the types of `d3/scale` ?

Reply:
Precisely:

![image](https://user-images.githubusercontent.com/2936402/199454142-d2cc9288-4820-49d9-9786-f46b1e13e4fb.png)

... but the code allows for a `function` and calls it: https://github.com/d3/d3-scale/blob/main/src/log.js#L121

Reply:
I would consider improving `@types/d3-format`, but since I'm considering bringing this logic in the codebase to have more control over it (for symlog, but also because the `scale.tickFormat` API feels super confusing).